packages/mui-material/src/Menu/Menu.js

```diff
@@ -198,22 +214,22 @@ const Menu = React.forwardRef(function Menu(inProps, ref) {
       open={open}
       ref={ref}
       transitionDuration={transitionDuration}
-      TransitionProps={{ onEntering: handleEntering, ...TransitionProps }}
+      TransitionProps={{ onEntering: handleEntering, ...transitionProps }}
+      TransitionComponent={slots.transition ?? TransitionComponent}
```

Comment:
The `Popover` component already implements `slots.transition` and `slotProps.transition`, we should use it.

Reply:
Based on the [latest code](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Popover/Popover.d.ts#L16-L22), there is no `slotProps.transition` yet.

Reply:
I will update this again when I work on Popover.

Reply:
My bad, I don't know why I though that it already had them.